platform/pwl.c

```diff
@@ -1218,74 +1308,37 @@ static void
 registry_global_remove (void *data, struct wl_registry *registry, uint32_t name)
 {
     PwlDisplay *display = data;
-    PwlOutput *item, *tmp;
-    wl_list_for_each_safe (item, tmp, &display->outputs, link) {
-        if (item->output_name == name) {
+    PwlOutput *output, *tmp_output;
+    wl_list_for_each_safe (output, tmp_output, &display->outputs, link) {
+        if (output->output_name == name) {
             g_debug ("Output #%"PRIi32" @ %p: Removed.",
-                     item->output_name, item->output);
-            g_clear_pointer (&item->output, wl_output_release);
-            wl_list_remove (&item->link);
-            g_slice_free (PwlOutput, item);
+                     output->output_name, output->output);
+            g_clear_pointer (&output->output, wl_output_release);
+            wl_list_remove (&output->link);
+            g_slice_free (PwlOutput, output);
             break;
         }
     }
+    PwlSeat *seat, *tmp_seat;
+    wl_list_for_each_safe (seat, tmp_seat, &display->seats, link) {
+        if (seat->seat_name == name) {
+            g_debug ("Seat #%"PRIi32" @ %p: Removed.",
+                     seat->seat_name, seat->seat);
+            g_clear_pointer (&seat->seat, wl_seat_release);
+            wl_list_remove (&seat->link);
+            g_slice_free (PwlSeat, seat);
```

Comment:
Just call `pwl_seat_destroy()` here. Also add a `break` because once we have found the item we want to remove from the list we don't need to continue iterating over it.

Reply:
done